### PR TITLE
fixing upload demux fastq

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qtp_target_gene/validate.py [hunk 1]
```diff
@@ -307,17 +308,20 @@ def _validate_demux_file(qclient, job_id, prep_info, out_dir, demux_fp,
...
-        demux_fp = new_demux_fp
-        fastq_fp = None
-        fasta_fp = None
+        # this if is important so we don't regenerate the demux file if the
+        # user uploads fastq or fna
+        if demux_fp != new_demux_fp:
+            copy(demux_fp, new_demux_fp)
+            demux_fp = new_demux_fp
+
+        if bool(id_map):
```
Comment (josenavas) [inline qtp_target_gene/validate.py]: I don't think this 'if' is required, if this point in the code is reached, this will always evaluate to True.

**Review by josenavas — APPROVED** [review]

File: qtp_target_gene/tests/test_validate.py [hunk 2]
```diff
@@ -512,6 +512,29 @@ def test_validate_demux_file(self):
...
+        files = {'preprocessed_fastq': [fastq_fp]}
+        job_id = self._create_template_and_job(
+            prep_info, files, "Demultiplexed")
+        obs_success, obs_ainfo, obs_error = _validate_demultiplexed(
+            self.qclient, job_id, prep_info, files, out_dir)
+        self.assertEqual(obs_error, "")
+        self.assertTrue(obs_success)
+
+        # we are gonna just check that the demux file looks good and [0] is
+        # because this only return one element in the object list
```
Comment (ElDeveloper) [inline qtp_target_gene/tests/test_validate.py]: return -> returns

File: qtp_target_gene/validate.py [hunk 3]
```diff
@@ -397,7 +400,7 @@ def _validate_demultiplexed(qclient, job_id, prep_info, files, out_dir):
     elif fastq:
         # Generate the demux file from the fastq
         demux = join(out_dir, "%s.demux" % splitext(basename(fastq))[0])
-        with open_file(demux, "w") as f:
+        with File(demux, 'w') as f:
```
Comment (ElDeveloper) [inline qtp_target_gene/validate.py:403]: Too bad open_file doesn't support 'w' for HDF5 files.

**Review by ElDeveloper — APPROVED** [review]
One minor typo, aside from that it looks good.

File: dump.rdb [hunk 4]
```diff
@@ -0,0 +1 @@
+REDIS0006�ܳC�Z��V
```
Comment (ElDeveloper) [inline dump.rdb]: Can you remove this file, and maybe add a .gitignore entry if you think this might happen again?

**Comment by antgonza** [comment]
thanks for catching!

**Comment by ElDeveloper** [comment]
Thanks @antgonza !

**Pull request merged**